sql/job_api.sql

```diff
@@ -8,7 +8,8 @@ CREATE OR REPLACE FUNCTION @extschema@.add_job(
   config JSONB DEFAULT NULL,
   initial_start TIMESTAMPTZ DEFAULT NULL,
   scheduled BOOL DEFAULT true,
-  check_config REGPROC DEFAULT NULL
+  check_config REGPROC DEFAULT NULL,
+  fixed_schedule BOOL DEFAULT TRUE
```

Comment:
We had discussed with JF that we would like this new behavior to be the default while also allowing users to continue using the old one, thus the change. In latest-dev I have made it so the update sets the fixed_schedule to `false` for already existing jobs, thinking that would be enough to not break existing jobs behavior.

Reply:
@mkindahl wouldn't this only affect users that add **new** jobs by using an existing script (the existing format of `add_job`)?

If my understanding is correct and this update is not affecting existing jobs, is your concern about users dynamically setting new jobs or TimescaleDB servers? Or is this about a type of workflow that I am not thinking about right now?

Could you help me understand the use case a little bit better?

It feels to me that this not a breaking change per se; even though it could potentially break some (automated) workflows, it does not break existing system behavior. It seems to me like a smooth way to transition people to a new behavior without having to deprecate and remove existing functionality early. But I would welcome an explanation that could help me see this differently :-)

Reply:
It is correct that existing jobs will still use a drifting schedule and that newly created jobs will then use a fixed schedule.

The use-case I have in mind is when users are deploying new servers using, for example Ansible (or any other of the existing "infrastructure as code" platform like Terraform). In this case, there will be a bunch of prepared scripts that are executed to deploy a new server. If users have invested time in tailoring this to handle the fact that our jobs drift, defaulting to using a fixed schedule is a change in behavior and will break their workflow.

In general, breaking semantics is something you try to limit to major releases, which is why I suggest that we stick with a default that do not risk breaking any workflows. It adds the complication that you need to explicitly enable the fixed schedule, but I feel this is a minor inconvenience.

I still think that we should change the default to use fixed schedule going forward, but this might be better to do in a major release.

Reply:
Thank you @mkindahl, I agree that breaking semantics should be treated and announced as deprecations (similar to breaking changes), with their effect taking place on the next major release. 

What I am missing in this case, is how users would set their workflows based on the drifting semantics, given that they are non deterministic (unpredictable on how the start times move forward in time). Other than canceling and redeploying jobs, I can't see myself depending any business logic on a scheduler with drifting semantics.

In my past experience, I have implemented job systems on top of drifting semantics only in two cases: 
- with very sort scheduling intervals; for jobs that you want to almost constantly run, just giving the system a little bit of time to breathe between executions. (in those cases the jobs are usually short lived as well, but this is not a strict requirement)
- for maintenance jobs that you don't really care about when exactly they are going to run, just that they will run a specific amount of time after their effects take place.

In both cases I can not see any semantics breaking, as there are no assumptions I can make while building my system on top of the drifting semantics, at least none that would break once the system would switch to fixed schedule semantics.

But this may be my experience bias and my lack of knowledge for other or more advanced use cases, hence my attempt to better understand the user workflows that would break.

Reply:
Summary of our discussion in the Team call and the final decision on how to move forward:

We are going to keep the `fixed_schedule` default to `TRUE` 

Additional reasons discussed:
- Our documentation on [add_job()](https://docs.timescale.com/api/latest/actions/add_job/) never mentions the drift semantics. In contrast, we present the behavior as a non drifting one:

   > Interval _between executions_ of this job

   Following discussions on our public slack, our drifting behavior seems to be unexpected, catching users by surprise. With this update we update the behavior to match our docs.

- Same for our `add_XXX_policy()` functions, which either don't mention the scheduling behavior or reference the interval between executions.

- Even if there is a way existing users' workflows to somehow be affected, there is a clear path on how to mitigate the issue with an one time fix.

/cc @konskov